app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,16 @@
+class SessionsController < ApplicationController
+  def create
+    @user = User.find_or_create_from_auth(auth)
+    if @user
+      session[:user_id] = @user.id
+      redirect_to dashboard_path
+    else
+      redirect_to root_path
+    end
+  end
+
+  private
+  def auth
+    request.env['omniauth.auth']
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.